### DISCUSSION TITLE
Terrain data for Portugal

### DISCUSSION THREAD

**Discussion by JGBotella** [BuildTheEarth/terraplusplus#65] [Q&A]
Hello, idk if this is the right place to ask but I'm currently trying to create a map from a region in Portugal for use in a server. My issue is I can't find a high resolution dataset (max I've been able to find is 20m per pixel) to base the map on. BTE/T++ has great quality terrain data/height maps that are more in line with what I'm looking for and the "Individual guides for countries" page directed me to seemingly defunct discord links and here, so I thought I'd go ahead and ask here. Thanks in advance.

**Comment by SmylerMC** [comment]
Terraplusplus uses various datasets depending on the location. As far as I can tell, Portugal is using the defaut AWS terrain tiles which aren't any better than what you are describing.

Elevation data sources are described in [this file](https://github.com/BuildTheEarth/terraplusplus/blob/5ae750171a0c6b8ef3cdf586b9dafe9851c84e53/src/main/resources/net/buildtheearth/terraplusplus/dataset/scalar/heights.json5).

**Comment by DavixDevelop** [comment]
The `Individual guides for countries`is/was meant for countries with a specific preparation of It's dataset to be be converted to a custom terrain. Guides were added on a case by case basis, if a user requested help with their specific dataset, hence the `contact me` in the footer of each guide/sub-guide. If the help fell outside the typical guide for custom terrain dataset preparation, ergo custom scripts needed to be written, I fully took over the task of creating the custom terrain dataset, and wrote the guide afterwards. 

Now regarding your question, most of Europe's terrain tiles are based on a dataset of 30m resolution, as seen here in these [datasets](https://github.com/tilezen/joerd/blob/master/docs%2Fdata-sources.md) that TerraPlusPlus uses in combination with custom hosted dataset, which Portugal isn't part of yet, so in game, the area uses the terrain tiles based on the 30m dataset.
